home/export_ordered_sets.py

```diff
@@ -0,0 +1,172 @@
+
+    def perform_export(self) -> Iterable[ExportRow]:
+        for item in self.queryset:
+            yield ExportRow(
+                name=item.name,
+                profile_field=item.profile_field,
+                page_slugs=item.page_slugs,
+                page=item.page,
+                unit=item.unit,
+                before_or_after=str(item.before_or_after),
+                contact_field=item.contact_field,
+            )
+
+
+def serialize_list(items: Iterable[str]) -> str:
+    """
+    Uses CSV formatting to seralize a list of strings, handling escaping
+    """
+    output = io.StringIO()
+    writer = csv.writer(output)
+    writer.writerow(items)
+    return output.getvalue().rstrip("\r\n")
+
+
+class OrderedSetsExportWriter:
+    rows: Iterable[ExportRow]
+
+    def __init__(self, rows: Iterable[ExportRow]):
+        self.rows = rows
+
+    def write_xlsx(self, response: HttpResponse) -> None:
+        workbook = Workbook()
+        worksheet: Worksheet = workbook.active  # type: ignore  # This is not a write or read only workbook
+        worksheet.append(ExportRow.headings())
+        for row in self.rows:
+            worksheet.append(row.to_tuple())
+        _set_xlsx_styles(workbook, worksheet)
+        workbook.save(response)
+
+    def write_csv(self, response: HttpResponse) -> None:
+        writer = csv.DictWriter(f=response, fieldnames=ExportRow.headings())
+        writer.writeheader()
+
+        for row in self.rows:
+            writer.writerow(row.to_dict())
+
+
+def _set_xlsx_styles(wb: Workbook, sheet: Worksheet) -> None:
+    """
+    Sets the style for the workbook adding any formatting that will make the sheet more
+    aesthetically pleasing
+    """
+    # Adjustment is because the size in openxlsx and google sheets are not equivalent
+    adjustment = 7
+
+    # Set columns based on best size
+    column_widths_in_pts = {
+        "title": 110,
+        "question_type": 110,
+        "tags": 110,
+        "slug": 110,
+        "version": 90,
+        "locale": 50,
+        "high_result_page": 110,
+        "high_inflection": 110,
+        "medium_result_page": 120,
+        "medium_inflection": 110,
+        "low_result_page": 110,
+        "skip_threshold": 110,
+        "skip_high_result_page": 110,
+        "generic_error": 370,
+        "question": 370,
+        "explainer": 370,
+        "error": 370,
+        "min": 110,
+        "max": 110,
+        "answers": 370,
+        "scores": 110,
+        "answer_semantic_ids": 110,
+        "question_semantic_id": 110,
+    }
+    for column in sheet.iter_cols(max_row=1):
+        [cell] = column
+        width = column_widths_in_pts[str(cell.value)]
+        sheet.column_dimensions[get_column_letter(cell.col_idx)].width = ceil(
+            width / adjustment
+        )
+
+    # Freeze heading row and side panel, 1 added because it freezes before the column
+    panel_column = get_column_letter(5)
+    sheet.freeze_panes = sheet[f"{panel_column}2"]
```

Comment:
I don't think there's any side panels in this, I think this will just freeze the name and profile fields. I don't think we need any frozen panels for ordered content sets, it's not a very wide table.